t/jv-allof.t

```diff
@@ -39,4 +39,11 @@ validate_ok 'he110th3re', $schema,
   E('/', '/oneOf/1 String is too long: 10/4.'),
   E('/', '/oneOf/1 String does not match ^[a-z]+$.');
 
+
+validate_ok 'hello', {type => ['integer', 'boolean']},
+  E('/', 'Expected integer/boolean - got string.');
```

Comment:
I think the message should have the types sorted, despite the order in `type`. (See #198)

Reply:
I disagree!  Any time there is a defined order in the schema (an arrayref, vs a hashref) I think we should keep the order as presented by the user.

Reply:
Most people don’t care about order (based on experience working with people) when they write their schemas, so we don’t either when creating the error messages.

Reply:
I have not changed the sorting in this PR. Let's look at that later.